### PR TITLE
Feature/profile

### PR DESCRIPTION
**Pull request by dyxcooh**
Adds a profile activity which lets the user edit their name, bio, and profile picture.
Implements the MVVM pattern.
Profile is synced with Firestore and the picture is stored to Firebase Storage. The default profile picture is the Meili logo.

### CODE REVIEW

**Comment by MarcelTorne** [comment]
Thanks for your work! I wanted to ask you for some changes before I revise, I see that there are some conflicts and that GoogleSignInActivity and AuthenticationService are not in the correct folder. Maybe it you don't have the latest changes in your branch and you should rebase with main again? Thanks in advance

File: app/src/main/java/com/github/epfl/meili/ContainerActivity.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.github.epfl.meili
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+
+class ContainerActivity : AppCompatActivity() {
```
[the remark this answers is not in the document]
Comment (dyxcooh) [reply app/src/main/java/com/github/epfl/meili/ContainerActivity.kt:6]: I thought at the beginning that with the profile as fragment it is possible to integrate other fragment, for example setting, but it's not necessary, maybe is better to switch to an activity in order to simplify

File: app/src/main/java/com/github/epfl/meili/AppGlideModule.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import com.bumptech.glide.Registry
+import com.bumptech.glide.annotation.GlideModule
+import com.bumptech.glide.module.AppGlideModule
+import com.firebase.ui.storage.images.FirebaseImageLoader
+import com.google.firebase.storage.StorageReference
+import java.io.InputStream
+
+
+@GlideModule
+class AppGlideModule: AppGlideModule() {
```
[the remark this answers is not in the document]
Comment (dyxcooh) [reply app/src/main/java/com/github/epfl/meili/AppGlideModule.kt]: Indeed, I intended to use this class to load profile photo, but it created some dependencies errors, and I found out that I can use an existant Glide library to bypass the problem, so I no longer use it

**Review by Bwananah — APPROVED** [review]
Good job ! Thanks for your work !

**Review by Bwananah — APPROVED** [review]

**Review by MarcelTorne — APPROVED** [review]
Thanks @dyxcooh for the work and thanks @aman3014 too for this big push! We are finally merging this feature !🎉

**Pull request merged**